scripts/gen/tpcc.sh

```diff
@@ -116,232 +75,31 @@ exec &> >(tee -a "$logdir/script.log")
-#then
-#  f_inc=$f_warehouses
-#fi
-
-# We first run an exploration test round with binary search to find a right
-# range of test boundary to use in the subsequent official run.
-# For certain machine types the binary search result don't work well and will
-# cause kv errors and end the test, so we need to handle them differently.
-tpcc_files=""
-f_inc=50
-has_good_result=0
-if [[ $special_type -eq 0 ]];
-then
-  l=$f_active
-  r=$f_warehouses
-  while [[ $l -le $r ]]
-  do
-      mid=$((l + ((r - l)/2)))
-      report="${logdir}/tpcc-results-$mid.txt"
-      ./cockroach workload run tpcc \
-          --warehouses="$mid"  \
-          --active-warehouses="$mid" \
-          --ramp=1m --duration="1m" \
-          "${pgurls[@]}" > "$report"
-
-      # awk uses lexicographic order by default when doing comparison, we
-      # need to force it to use number comparison.
-      prev_line=$(tail -2 "$report")
-      if [[ "$prev_line" == *efc* ]] && [[ $(tail -1 "$report" | awk '{if(int($3) > 87){print "pass"}}') == "pass" ]];
-      then
-        echo "Test TPCC on $mid failed"
-        ((r=mid-f_inc))
-        current_result_good=0
-      else
-        echo "Test TPCC on $mid passed"
-        ((l=mid+f_inc))
-        current_result_good=1
-        if [[ $mid -gt $best_result ]];
-        then
-          best_result=$mid
-        fi
-      fi
-  done
+  #./cockroach sql --insecure --url "${pgurls[0]}" -e "
+  # SET CLUSTER SETTING kv.snapshot_recovery.max_rate = '512 MiB';
+  # SET CLUSTER SETTING kv.snapshot_rebalance.max_rate = '512 MiB';
+	# SET CLUSTER SETTING admission.kv.enabled,=false;
+	# SET CLUSTER SETTING admission.sql_kv_response.enabled=false;
+	# SET CLUSTER SETTING admission.sql_sql_response.enabled=false;
+  #";
+  echo "Loading TPCC fixture for $f_warehouses warehouses ..."
+  ./cockroach workload fixtures load tpcc --checks=false --warehouses="$f_warehouses" "$f_load_args" "${pgurls[0]}"
+  echo "done loading"
 fi
 
-result_files=(`find "${logdir}"/ -maxdepth 1 -name "tpcc-results*.txt"`)
-if [ ${#result_files[@]} -gt 0 ];
+if (( f_active == 0 ))
 then
-  rm "${logdir}"/tpcc-results*.txt
-  echo "Deleted exploration run result."
-fi
-
-
-# if valid result is found in the previous test round, we start an official
-# run after adjusting of the starting point to a multiple of one hundred.
-# Because at present tpcc is pretty unstable where result can fluctuate
-# within quite a range, we retry a few times with lower boundary for the
-# official run stage before we call the test fail.
-#
-# In order to achieve better comparison result across multiple machine
-# types, we use a small increase of store size of 50.
-if [ $best_result -gt 0 ]
-then
-  ((l=$best_result-$best_result%f_inc))
-  if [ $current_result_good -eq 0 ]
+  # Scale active warehouse count by f_active_per_core * number of CPUs.
+  f_active=$(( f_active_per_core * $(cat /proc/cpuinfo | grep processor | wc -l) ))
+  if (( f_active_per_core > f_warehouses ))
```

Comment:
I don't quite understand this condition -- did you mean `if (( f_active> f_warehouses ))`?
nit: maybe rename `f_active` to `f_active_warehouses`.

Reply:
Copy pasta.